ConcurrentModificationException while reading System Properties

If a the System Properties are updated while they are being iterated over, a ConcurrentModificationException may occur;
```
Caused by: java.util.ConcurrentModificationException
    at java.util.Hashtable$Enumerator.next(Hashtable.java:1506)
    at java.util.Iterator.forEachRemaining(Iterator.java:127)
    at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1812)
    at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:523)
    at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:513)
    at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:719)
    at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:245)
    at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:510)
    at io.smallrye.config.common.utils.ConfigSourceUtil.propertiesToMap(ConfigSourceUtil.java:48)
    at io.smallrye.config.SysPropConfigSource.getSystemProperties(SysPropConfigSource.java:57)
    at io.smallrye.config.SysPropConfigSource.getProperties(SysPropConfigSource.java:43)
    at io.smallrye.config.SysPropConfigSource.getPropertyNames(SysPropConfigSource.java:48)
    at io.smallrye.config.ConfigValueConfigSourceWrapper.getPropertyNames(ConfigValueConfigSourceWrapper.java:53)
    at io.smallrye.config.SmallRyeConfigSourceInterceptor.iterateNames(SmallRyeConfigSourceInterceptor.java:34)
    at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.iterateNames(SmallRyeConfigSourceInterceptorContext.java:25)
    at io.smallrye.config.SmallRyeConfigSourceInterceptor.iterateNames(SmallRyeConfigSourceInterceptor.java:33)
    at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.iterateNames(SmallRyeConfigSourceInterceptorContext.java:25)
    at io.smallrye.config.SmallRyeConfigSourceInterceptor.iterateNames(SmallRyeConfigSourceInterceptor.java:33)
    at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.iterateNames(SmallRyeConfigSourceInterceptorContext.java:25)
    at io.smallrye.config.ConfigSourceInterceptor.iterateNames(ConfigSourceInterceptor.java:58)
    at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.iterateNames(SmallRyeConfigSourceInterceptorContext.java:25)
    at io.smallrye.config.ConfigSourceInterceptor.iterateNames(ConfigSourceInterceptor.java:58)
    at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.iterateNames(SmallRyeConfigSourceInterceptorContext.java:25)
    at io.smallrye.config.ProfileConfigSourceInterceptor.iterateNames(ProfileConfigSourceInterceptor.java:93)
    at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.iterateNames(SmallRyeConfigSourceInterceptorContext.java:25)
    at io.smallrye.config.ConfigSourceInterceptor.iterateNames(ConfigSourceInterceptor.java:58)
    at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.iterateNames(SmallRyeConfigSourceInterceptorContext.java:25)
    at io.smallrye.config.SmallRyeConfig.getPropertyNames(SmallRyeConfig.java:240)
    at io.smallrye.config.inject.ConfigExtension.validate(ConfigExtension.java:127)
```

Since Properties is based on Hashtable, all methods are synchronized, so adding an extra synchronized block should not be any performance problem. Will submit a PR.

--- a/common/src/main/java/io/smallrye/config/common/utils/ConfigSourceUtil.java
+++ b/common/src/main/java/io/smallrye/config/common/utils/ConfigSourceUtil.java
@@ -46,8 +46,10 @@ public class ConfigSourceUtil {
      */
     public static Map<String, String> propertiesToMap(Properties properties) {
         Map<String, String> map = new HashMap<>();
-        for (Map.Entry<Object, Object> e : properties.entrySet()) {
-            map.put(String.valueOf(e.getKey()), String.valueOf(e.getValue()));
+        synchronized (properties) {
+            for (Map.Entry<Object, Object> e : properties.entrySet()) {
+                map.put(String.valueOf(e.getKey()), String.valueOf(e.getValue()));
+            }
         }
         return map;
     }

